nixos/doc/manual/development/option-declarations.xml

```diff
@@ -65,4 +65,68 @@ options = {
 
 </para>
 
+<section xml:id="sec-option-declarations-eot"><title>Extended Option 
+    Types</title>
+
+  <para>Extended option types is the ability to extend certain types 
+    declaration through multiple module files.
+    This feature only work with a restricted set of types, namely 
+    <literal>enum</literal> and its composed forms.</para>
+
+  <para>Extended option types can be used for <literal>enum</literal> options 
+    that affects multiple modules, or as an alternative to related 
+    <literal>enable</literal> options.</para>
+
+  <para>As an example, we will take the case of a service 
+    <literal>foo</literal>, that has multiple backends: <literal>bar</literal> 
+    and <literal>baz</literal>.
```

Comment:
foo, bar, and baz might not be easy to follow, maybe taking a concrete example might help get a better understanding of the feature provided by extensible option types.

For examples:
 * service: audio; backends: alsa, pulseaudio.
 * service: webserver; backends: apache, nginx.
 * service: init-system; backends: upstart, systemd, initd.
 * service: display manager; backends: kdm, slim.